pex/pex_bootstrapper.py

```diff
@@ -18,122 +18,161 @@
 __all__ = ('bootstrap_pex',)
 
 
-def find_in_path(target_interpreter):
-  if os.path.exists(target_interpreter):
-    return target_interpreter
-
-  for directory in os.getenv('PATH', '').split(os.pathsep):
-    try_path = os.path.join(directory, target_interpreter)
+def _find_pex_python(pex_python):
+  def try_create(try_path):
     if os.path.exists(try_path):
-      return try_path
+      try:
+        return PythonInterpreter.from_binary(try_path)
+      except Executor.ExecutionError:
+        pass
+    return None
```

Comment:
Ah did no realize this `pass` was originally nested in the `if`. My bad. This is getting a little nit picky, but I think this would be the most explicit:

```python
if not os.path.exists(try_path):
  return None
try:
  returnPythonInterpreter.from_binary(try_path)
except Executor.ExecutionError:
  return None
```

Reply:
Erm, OK. This is truly in the eye of the beholder. I killed the path existence check though and that swings this back around to reasonable to me. Done-ish.